src/orange/surf/Involute.hh

```diff
@@ -0,0 +1,305 @@
+    using constants::pi;
+
+    real_type x = pos[0] - origin_[0];
+    real_type const y = pos[1] - origin_[1];
+
+    if (this->sign() == Sign::clockwise)
+    {
+        x = negate(x);
+    }
+
+    // Calculate distance to origin and obtain t value for distance.
+    real_type const t_point_sq = (ipow<2>(x) + ipow<2>(y) / ipow<2>(r_b_)) - 1;
+
+    // Check if point is in defined bounds.
+    if (t_point_sq < ipow<2>(tmin_))
+    {
+        return SignedSense::outside;
+    }
+    if (t_point_sq > ipow<2>(tmax_))
+    {
+        return SignedSense::outside;
+    }
+
+    // Check if Point is on involute.
+    detail::InvolutePoint calc_point{this->r_b(), a_};
+    Real2 point = calc_point(clamp_to_nonneg(t_point_sq));
+
+    if (x == point[0] && y == point[1])
+    {
+        return SignedSense::on;
+    }
+
+    // Check if point is in interval
+
+    // Calculate tangent point
+    real_type x_prime = ipow<2>(r_b_) / std::sqrt(ipow<2>(x) + ipow<2>(y));
+    real_type y_prime = std::sqrt(ipow<2>(r_b_) - ipow<2>(x_prime));
+
+    point[0] = (x_prime * x - y_prime * y) / std::sqrt(ipow<2>(x) + ipow<2>(y));
+    point[1] = (y_prime * x + x_prime * y) / std::sqrt(ipow<2>(x) + ipow<2>(y));
+
+    // Calculate angle of tangent
+    real_type theta = std::acos(point[0] / norm(point));
+    if (point[1] < 0)
+    {
+        theta = 2 * pi - theta;
+    }
+    theta += max<real_type>(0, std::floor((tmax_ + a_ - theta) / (2 * pi))) * 2
+             * pi;
+
+    // Calculate the displacement angle of the point
+    real_type a1 = theta - std::sqrt(clamp_to_nonneg(t_point_sq));
+
+    // Check if point is inside bounds
+    if (theta < tmax_ + a_ && a1 >= a_)
+    {
+        return SignedSense::inside;
+    }
+
+    return SignedSense::outside;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Calculate all possible straight-line intersections with this surface.
+ */
+CELER_FUNCTION auto
+Involute::calc_intersections(Real3 const& pos,
+                             Real3 const& dir,
+                             SurfaceState on_surface) const -> Intersections
+{
+    // Expand translated positions into 'xyz' coordinate system
+    Real3 rel_pos{pos};
+    rel_pos[0] -= origin_[0];
+    rel_pos[1] -= origin_[1];
+
+    detail::InvoluteSolver solve(this->r_b(), a_, this->sign(), tmin_, tmax_);
+
+    return solve(rel_pos, dir, on_surface);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Calculate outward normal at a position on the surface.
+ *
+ * This is done by taking the derivative of the involute equation : \f[
+ * n = {\sin(t+a) - \cos(t+a), 0}
+ * \f]
+ */
+CELER_FORCEINLINE_FUNCTION Real3 Involute::calc_normal(Real3 const& pos) const
+{
+    real_type const x = pos[0] - origin_[0];
+    real_type const y = pos[1] - origin_[1];
+
+    // Calculate normal
+    real_type const angle
+        = std::sqrt((ipow<2>(x) + ipow<2>(y) / (ipow<2>(r_b_))) - 1) + a_;
```

Comment:
```suggestion
        = std::sqrt((ipow<2>(x) + ipow<2>(y)) / ipow<2>(r_b_) - 1) + a_;
```
You're sure it's not this?  Again, if so, it probably needs another test or two.